test/unit/lib/system/deploy_test.rb

```diff
@@ -14,11 +15,32 @@ class DeployTest < ActiveSupport::TestCase
     assert_equal 'x', System::Deploy.info.minor_version
   end
 
-  test 'custom release' do
-    path = Rails.root.join('test', 'fixtures', 'deploy_info').expand_path
-    System::Deploy.load_info! ActiveSupport::JSON.decode(path.read)
+  class OnpremisesDeployTest < DeployTest
+    setup do
+      ThreeScale.config.stubs(onpremises: true)
+    end
+
+    test 'custom release' do
+      path = Rails.root.join('test', 'fixtures', 'deploy_info').expand_path
+      System::Deploy.load_info! ActiveSupport::JSON.decode(path.read)
 
-    assert_equal '4', System::Deploy.info.major_version
-    assert_equal '5', System::Deploy.info.minor_version
+      assert_equal '4', System::Deploy.info.major_version
+      assert_equal '5', System::Deploy.info.minor_version
+    end
   end
+
+  class SaasDeployTest < DeployTest
+    setup do
+      ThreeScale.config.stubs(onpremises: false)
+    end
+
+    test 'custom release from .deploy_info is ignored' do
+      path = Rails.root.join('test', 'fixtures', 'deploy_info').expand_path
+      System::Deploy.load_info! ActiveSupport::JSON.decode(path.read)
+
+      assert_equal '2', System::Deploy.info.major_version
+      assert_equal 'x', System::Deploy.info.minor_version
```

Comment:
maybe read these from their sources

Reply:
Not sure I understand... what sources?

Reply:
I mean from the `VERSION` constant, not hardcode here. But that's a nitpick

Reply:
Ah, I see. Well, I actually prefer to use values in the tests rather than constants, as it is more explicit.
If somebody changes the constant accidentally, the test will catch it.
This of course only works for this case, where the version is stable and not expected to be change a lot (if ever).